scene/animation/tween.cpp

```diff
@@ -797,16 +798,50 @@ void Tween::_remove(Object *p_object, StringName p_key, bool first_only) {
 			continue;
 		if (object == p_object && (data.concatenated_key == p_key || p_key == "")) {
 			for_removal.push_back(E);
-			if (first_only) {
-				break;
-			}
 		}
 	}
 	for (List<List<InterpolateData>::Element *>::Element *E = for_removal.front(); E; E = E->next()) {
 		interpolates.erase(E->get());
 	}
 }
 
+void Tween::_remove_by_uid(int uid) {
+	if (pending_update != 0) {
+		call_deferred("_remove_by_uid", uid);
+		return;
+	}
+
+	for (List<InterpolateData>::Element *E = interpolates.front(); E; E = E->next()) {
+		if (uid == E->get().uid) {
+			E->erase();
+			break;
+		}
+	}
+}
+
+void Tween::_push_interpolate_data(InterpolateData &p_data) {
+	pending_update++;
+	int uid = 0;
+	if (!interpolates.empty()) {
+		// We are looking for the first available uid
+		while (true) {
```

Comment:
This is going to be a performance-intensive loop if you make many tweens, since it is going to loop N * N / 2 times for N tweens if none are removed.

It might be better if the tween keeps an `private: int uid` field, and increments it every time. The int might wraparound, but it is expected when running ~2¹⁶ tweens at once.

Reply:
Yes, I was expecting a performance issue here.
`Tween::remove_all` is ok, but do you think we should try to find the lowest uid at the end of `Tween::_remove` & `Tween::_remove_by_uid`?

Reply:
No, since the lowest uid might have some uid-s high than it which are already taken. No one would make that many tweens on a single node as to overflow the int anyway.

Reply:
Ok, great!